components/resource_control/src/resource_group.rs

```diff
@@ -244,6 +247,16 @@ impl ResourceController {
 
         // maybe update existed group
         self.resource_consumptions.write().insert(name, group);
+        self.check_customized();
+    }
+
+    fn check_customized(&self) {
+        let groups = self.resource_consumptions.read();
+        if groups.len() == 1 && groups.get(DEFAULT_RESOURCE_GROUP_NAME.as_bytes()).is_some() {
```

Comment:
NIT: Our logic always ensures that the "default" group exists, so no need this extra check. But this is ok to me since this is not in the hot path.